mmv1/third_party/terraform/services/bigtable/resource_bigtable_authorized_view.go

```diff
@@ -0,0 +1,462 @@
+				Optional:     true,
+				Computed:     true,
+				ValidateFunc: validation.StringInSlice([]string{"PROTECTED", "UNPROTECTED"}, false),
+				Description:  `A field to make the authorized view protected against data loss i.e. when set to PROTECTED, deleting the authorized view, the table containing the authorized view, and the instance containing the authorized view would be prohibited. If not provided, currently deletion protection will be set to UNPROTECTED as it is the API default value.`,
+			},
+
+			"subset_view": {
+				Type:        schema.TypeList,
+				Optional:    true,
+				MaxItems:    1,
+				Description: `An AuthorizedView permitting access to an explicit subset of a Table.`,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"row_prefixes": {
+							Type:        schema.TypeSet,
+							Optional:    true,
+							Elem:        &schema.Schema{Type: schema.TypeString},
+							Description: `Base64-encoded row prefixes to be included in the authorized view. To provide access to all rows, include the empty string as a prefix ("").`,
+						},
+						"family_subsets": {
+							Type:        schema.TypeSet,
+							Optional:    true,
+							Description: `Subsets of column families to be included in the authorized view.`,
+							Elem:        familySubsetSchemaElem,
+						},
+					},
+				},
+			},
+		},
+		UseJSONNumber: true,
+	}
+}
+
+func resourceBigtableAuthorizedViewCreate(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*transport_tpg.Config)
+	userAgent, err := tpgresource.GenerateUserAgentString(d, config.UserAgent)
+	if err != nil {
+		return err
+	}
+
+	ctx := context.Background()
+
+	project, err := tpgresource.GetProject(d, config)
+	if err != nil {
+		return err
+	}
+
+	instanceName := tpgresource.GetResourceNameFromSelfLink(d.Get("instance_name").(string))
+	c, err := config.BigTableClientFactory(userAgent).NewAdminClient(project, instanceName)
+	if err != nil {
+		return fmt.Errorf("Error starting admin client. %s", err)
+	}
+	if err := d.Set("instance_name", instanceName); err != nil {
+		return fmt.Errorf("Error setting instance_name: %s", err)
+	}
+
+	defer c.Close()
+
+	authorizedViewId := d.Get("name").(string)
+	tableId := d.Get("table_name").(string)
+	authorizedViewConf := bigtable.AuthorizedViewConf{
+		AuthorizedViewID: authorizedViewId,
+		TableID:          tableId,
+	}
+
+	// Check if deletion protection is given
+	// If not given, currently tblConf.DeletionProtection will be set to false in the API
+	deletionProtection := d.Get("deletion_protection")
+	if deletionProtection == "PROTECTED" {
+		authorizedViewConf.DeletionProtection = bigtable.Protected
+	} else if deletionProtection == "UNPROTECTED" {
+		authorizedViewConf.DeletionProtection = bigtable.Unprotected
+	}
+
+	subsetView, ok := d.GetOk("subset_view")
+	if !ok || len(subsetView.([]interface{})) != 1 {
+		return fmt.Errorf("subset_view must be specified for authorized view %s", authorizedViewId)
```

Comment:
Why do we have validation for `subset_view` here, but have it set as optional in the field schema?

Reply:
That makes sense. Refreshing to see considerations for future changes, thank you!